lib/ex_aws/dynamo.ex

```diff
@@ -596,6 +591,103 @@ defmodule ExAws.Dynamo do
+          {:return_consumed_capacity, return_consumed_capacity_vals}
+        ]
+
+  @spec transact_get_items(items :: [transact_get_item], transact_get_items_opts) :: ExAws.Operation.JSON.t()
+  @spec transact_get_items(items :: [transact_get_item]) :: ExAws.Operation.JSON.t()
+  def transact_get_items(items, opts \\ []) do
+    data =
+      opts
+      |> build_opts
+      |> Map.merge(%{
+        "TransactItems" => Enum.map(items, &build_transaction_item({:get, &1}))
+      })
+
+    request(:transact_get_items, data)
+  end
+
+  defp build_transaction_item({method, {table, item}}), do: build_transaction_item({method, {table, item, []}})
+
+  defp build_transaction_item({method, {table, item, opts}}) do
+    build_transaction_item(method, table, item, opts)
+  end
+
+  defp build_transaction_item(method, table_name, item, opts) do
+    item = item |> Map.new() |> Dynamo.Encoder.encode_root()
+
+    details =
+      opts
+      |> build_opts()
+      |> Map.merge(%{
+        "TableName" => table_name,
+        transaction_item_key(method) => item
+      })
+
+    %{camelize(method) => details}
+  end
+
+  defp transaction_item_key(:put), do: "Item"
+  defp transaction_item_key(_any), do: "Key"
+
+  @type return_values_on_condition_check_failure_vals :: :all_old | :none
+
+  @type transact_standard_item_opts :: [
+          {:condition_expression, binary}
+          | {:expression_attribute_names, expression_attribute_names_vals}
+          | {:expression_attribute_values, expression_attribute_values_vals}
+          | {:return_values_on_condition_check_failure, return_values_on_condition_check_failure_vals}
+        ]
+
+  @type transact_update_item_opts :: [
+          {:condition_expression, binary}
+          | {:expression_attribute_names, expression_attribute_names_vals}
+          | {:expression_attribute_values, expression_attribute_values_vals}
+          | {:return_values_on_condition_check_failure, return_values_on_condition_check_failure_vals}
+          | {:update_expression, binary}
+        ]
+
+  @type transact_write_item ::
+          {:condition_check, {table_name :: binary, key :: primary_key, transact_standard_item_opts}}
+          | {:delete, {table_name :: binary, key :: primary_key, transact_standard_item_opts}}
+          | {:put, {table_name :: binary, item :: map(), transact_standard_item_opts}}
+          | {:update, {table_name :: binary, key :: primary_key, transact_update_item_opts}}
+
+  @type transact_write_items_opts :: [
+          {:client_request_token, binary}
+          | {:return_consumed_capacity, return_consumed_capacity_vals}
+          | {:return_item_collection_metrics, return_item_collection_metrics_vals}
+        ]
+
+  @spec transact_write_items(items :: [transact_write_item], transact_write_items_opts) :: ExAws.Operation.JSON.t()
+  @spec transact_write_items(items :: [transact_write_item]) :: ExAws.Operation.JSON.t()
+  def transact_write_items(items, opts \\ []) do
```

Comment:
@kyleaa Need an @doc for `transact_write_items`, yes?

Reply:
Following the same approach, 

"a synchronous write operation that groups up to 25 action requests"

Reply:
Great, thank you.